lib/modules/versioning/unity3d/index.ts

```diff
@@ -0,0 +1,81 @@
+import semver from 'semver';
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'unity3d';
+export const displayName = 'Unity3D';
+export const urls = [
+  'https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html#version-define-expressions:~:text=characters%20are%20supported.-,Unity%20version%20numbers,-Current%20versions%20of',
+];
+export const supportsRanges = false;
+
+class Unity3dVersioningApi extends GenericVersioningApi {
+  private static readonly ReleaseStreamType = new Map([
+    ['Alpha', 'a'],
+    ['Beta', 'b'],
+    ['Patch', 'p'],
+    ['Experimental', 'x'],
+    ['Stable', 'f'],
+    ['Stable (China)', 'c'],
+  ]);
+  private static readonly ReleaseStreamTypeKeyOrder = Array.from(
+    Unity3dVersioningApi.ReleaseStreamType.values(),
+  );
+
+  protected _parse(version: string): GenericVersion | null {
+    const matches = regEx(
+      /^(?<Major>\d+)\.(?<Minor>\d+)\.(?<Patch>\d+)(?<ReleaseStream>\w)(?<Build>\d+)/gm,
+    ).exec(version);
+    if (!matches?.groups) {
+      return null;
+    }
+    const { Major, Minor, Patch, ReleaseStream, Build } = matches.groups;
+
+    const release = [
+      parseInt(Major),
+      parseInt(Minor),
+      parseInt(Patch) * 1000 + parseInt(Build),
+    ];
+    const stable = [
+      Unity3dVersioningApi.ReleaseStreamType.get('Stable'),
+      Unity3dVersioningApi.ReleaseStreamType.get('Stable (China)'),
+    ].includes(ReleaseStream);
+
+    let suffix = '';
+    if (version.match(/\([a-f0-9]+\)$/)) {
+      suffix = version.substring(1).substring(version.length - 2);
+    }
```

Comment:
You can capture this from the regex, that way we can be more forgiving regarding whitespace.

Reply:
Updated in fac18e522406962e7150c39c40445d7e3f33b4bd, however I noticed that every second test fails, unless I reset `.lastIndex` before using it:
- https://github.com/renovatebot/renovate/commit/fac18e522406962e7150c39c40445d7e3f33b4bd#diff-a80b8c624eeecc631e405dedad9d67f3e918b27d1157a67fe557bfc863374960R33
- https://github.com/renovatebot/renovate/commit/fac18e522406962e7150c39c40445d7e3f33b4bd#diff-a80b8c624eeecc631e405dedad9d67f3e918b27d1157a67fe557bfc863374960R49

This seems to stem from the fact, that the parser doesn't reset, now that it's global. Let me know if there's a better way to address this!